ui/packages/catalog-ui-search/src/main/webapp/js/widgets/cesium.polygon.js

```diff
@@ -9,127 +9,80 @@
-                'changed': 'updatePrimitive'
-            },
-            updatePrimitive: function() {
-                this.drawPolygon(this.model);
-            },
-            drawPolygon: function(model) {
-                var json = model.toJSON();
-                var isMultiPolygon =  ShapeUtils.isArray3D(json.polygon);
-                var polygons = isMultiPolygon ? json.polygon : [ json.polygon ];
-
-                var color = this.model.get('color');
-
-                // first destroy old one
-                if (this.primitive && !this.primitive.isDestroyed()) {
-                    this.options.map.scene.primitives.remove(this.primitive);
-                }
-
-                this.primitive = new Cesium.PolylineCollection();
-
-                (polygons || []).forEach(function(polygonPoints){
-                    if (!polygonPoints || polygonPoints.length < 3) {
-                        return;
-                    }
-                    if (polygonPoints[0].toString() !== polygonPoints[polygonPoints.length - 1].toString()) {
-                        polygonPoints.push(polygonPoints[0]);
-                    }
-                    this.primitive.add({
-                        width: 8,
-                        material: Cesium.Material.fromType('PolylineOutline', {
-                            color: color ? Cesium.Color.fromCssColorString(color) : Cesium.Color.KHAKI,
-                            outlineColor: Cesium.Color.WHITE,
-                            outlineWidth: 4
-                        }),
-                        id: 'userDrawing',
-                        positions: Cesium.Cartesian3.fromDegreesArray(_.flatten(polygonPoints))
-                    });
-                }.bind(this));
-
-                this.options.map.scene.primitives.add(this.primitive);
-            },
-            destroy: function() {
-                if (this.primitive) {
-                    this.options.map.scene.primitives.remove(this.primitive);
-                }
-                this.remove(); // backbone cleanup.
+
+const Cesium = require('cesium');
+const _ = require('underscore');
+const ShapeUtils = require('js/ShapeUtils');
+const Turf = require('@turf/turf');
+const DistanceUtils = require('js/DistanceUtils');
+
+const { GeometryRenderView, GeometryController } = require('./cesium.base.line');
+
+"use strict";
+
+const createBufferedPolygonPoints = (polygonPoints, model) => {
+    const width = DistanceUtils.getDistanceInMeters(model.toJSON().polygonBufferWidth, model.get('polygonBufferUnits')) || 1;
+
+    return Turf.buffer(Turf.lineString(polygonPoints), Math.max(width, 1), 'meters');
+}
+
+class PolygonRenderView extends GeometryRenderView {
+    constructor(options) {
+        super({...options, modelEvents: {'changed': 'updatePrimitive'}});
+
+        this.cameraMagnitude;
+        this.animationFrameId;
+
+        this.init({...options, events: 'change:polygon change:polygonBufferWidth, change:polygonBufferUnits'});
```

Comment:
Is there supposed to be a comma in the `events` string?

Reply:
lol, didn't scroll all the way to the right and thought this was meaning I need to add comas. Will remove. Thanks @jrnorth !